services/src/datasets/listing.rs

```diff
@@ -72,24 +71,50 @@ pub enum OrderBy {
 
 /// Listing of stored datasets
 #[async_trait]
-pub trait DatasetProvider: Send
+pub trait DatasetProvider<S: Session>: Send
     + Sync
     + MetaDataProvider<MockDatasetDataSourceLoadingInfo, VectorResultDescriptor, VectorQueryRectangle>
     + MetaDataProvider<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>
     + MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
-    + ProvenanceProvider
 {
     // TODO: filter, paging
     async fn list(
         &self,
-        // session: &S, // TODO: authorization
+        session: &S,
         options: Validated<DatasetListOptions>,
     ) -> Result<Vec<DatasetListing>>;
 
-    // TODO: is this method useful?
-    async fn load(
-        &self,
-        // session: &S, // TODO: authorization
-        dataset: &DatasetId,
-    ) -> Result<Dataset>;
+    async fn load(&self, session: &S, dataset: &DatasetId) -> Result<Dataset>;
+
+    async fn provenance(&self, session: &S, dataset: &DatasetId) -> Result<ProvenanceOutput>;
+}
+
+#[async_trait]
+pub trait ExternalDatasetProvider: Send
```

Comment:
Maybe we could add some notes about that we would need to specify authorization as a parameter independent of the concrete context.

Reply:
adab347397faa698db32ae904b15882856f6dc56